library/java/io/envoyproxy/envoymobile/engine/EnvoyEngineImpl.java

```diff
@@ -50,18 +50,31 @@ public void terminate() {
   /**
    * Run the Envoy engine with the provided yaml string and log level.
    *
+   * The envoyConfiguration is used to resolve the configurationYAML.
+   *
+   * @param envoyConfiguration The EnvoyConfiguration used to start Envoy.
    * @param configurationYAML The configuration yaml with which to start Envoy.
-   * @param logLevel           The log level to use when starting Envoy.
+   * @param logLevel          The log level to use when starting Envoy.
    * @return A status indicating if the action was successful.
    */
   @Override
-  public int runWithConfig(String configurationYAML, String logLevel) {
-    try {
-      return JniLibrary.runEngine(this.engineHandle, configurationYAML, logLevel);
-    } catch (Throwable throwable) {
-      // TODO: Need to have a way to log the exception somewhere.
-      return ENVOY_FAILURE;
+  public int runWithConfig(EnvoyConfiguration envoyConfiguration, String configurationYAML,
```

Comment:
Do we care about not exposing a way to run with a resolved yaml file and no configuration object?

Reply:
We're currently not exposing this functionality to the Kotlin consumers. As the API design, the `EngineBuilder` manages the `EnvoyConfiguration` detail internally

Reply:
Generally, I think this is fine; nothing stops someone from providing resolved yaml and not bothering to configure anything via the builder. That said, I'm curious - why not mirror the iOS API directly?

Reply:
Yeah, I was mostly wondering about parity.

Reply:
Doesn't this mirror https://github.com/envoyproxy/envoy-mobile/blob/3a2fdb74d67f7ccc60e5fd8bd34aa2ae47e72003/library/objective-c/EnvoyEngineImpl.m#L454-L456? --  I can change the method name and signature to better reflect that.

Reply:
I meant in the absence of running with a resolved yaml and no config object. Or do we not expose that in iOS?

Reply:
like: https://github.com/envoyproxy/envoy-mobile/blob/3a2fdb74d67f7ccc60e5fd8bd34aa2ae47e72003/library/objective-c/EnvoyEngineImpl.m#L474

Reply:
If we're missing that method I can add it but I would prefer to keep this PR as purely just to make things works for the test

Reply:
sg, can you open an issue?

Reply:
This is actually what we have now https://github.com/envoyproxy/envoy-mobile/blob/3a2fdb74d67f7ccc60e5fd8bd34aa2ae47e72003/library/objective-c/EnvoyEngine.h#L312-L331